### PR TITLE
Fix Bug 1482138 - Expose "Add search engine" string from search.properties for use in Search Shortcut editing menu

### PR DESCRIPTION
**Pull request by Mardak**
Fall back to existing translation of search.properties "Add Search Engine".

Blocked on the actual string id that @rlr is using.

### CODE REVIEW

**Comment by rlr** [comment]
I'm using `section_menu_action_add_search_shortcut` 👀  but I can change it to whatever.

**Comment by rlr** [comment]
`section_menu_action_add_search_engine` WFM

File: bin/strings-import.js [hunk 1]
```diff
@@ -44,8 +44,21 @@ async function saveProperties(locale) {
...
   }
 
+  // For now, detect if a string is missing to use a different one instead
+  let text = await response.text();
+  const expectedText = "section_menu_action_add_search_engine=";
+  if (!text.includes(expectedText)) {
+    const otherFile = await fetch(`${L10N_CENTRAL}/${locale}/raw-file/default/browser/chrome/browser/search.properties`);
+    if (otherFile.ok) {
+      const otherText = await otherFile.text();
+      const otherMatch = otherText.match(/searchAddFoundEngine2=([^\n]+)/);
```
[the remark this answers is not in the document]
Comment (Mardak) [reply bin/strings-import.js]: Thanks for the link. Looks like ja/-JP-mac were missing strings too:
https://hg.mozilla.org/l10n-central/ja/file/default/browser/chrome/browser/search.properties#l36

File: bin/strings-import.js [hunk 2]
```diff
@@ -34,6 +34,22 @@ async function getLocales() {
...
+      transvision[string] = response.ok ? await response.json() : {};
+    }
+    return transvision[string];
+  };
+  const expectedKey = "section_menu_action_add_search_engine";
+  const expected = await getTransvision(`browser/chrome/browser/activity-stream/newtab.properties:${expectedKey}`);
+  const target = await getTransvision("browser/chrome/browser/search.properties:searchAddFoundEngine2");
+  return !expected[locale] && target[locale] ? `${expectedKey}=${target[locale]}\n` : "";
+}
+
```
Comment (k88hudson) [inline bin/strings-import.js]: A+ hax

**Comment by Mardak** [comment]
I think this can land even before #4317? The key is indeed `section_menu_action_add_search_engine` there r? @k88hudson

**Pull request merged**